### PR TITLE
Add uv_dlerror() and uv_dlerror_free()

### PR DESCRIPTION
**Pull request by shigeki**
Per the discussion in #286, two new APIs are added to libuv in order to retrieves a detail error message of dynamic linking loaders such as dlopen(). 
Please review this.

### CODE REVIEW

File: test/test-dlerror.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+#include "uv.h"
+#include "task.h"
+#include <string.h>
+
+const char* path = "test/fixtures/load_error.node";
+const char* msg;
+
+#ifdef __linux__
+  const char* dlerror_desc = "file too short";
```
Comment (bnoordhuis) [inline test/test-dlerror.c:30]: I like how you've added a test but it'll fail with non-English locales.

**Comment by shigeki** [comment]
@bnoordhuis Thank you for your comment, Ben.
Actually, the default locale is "C" unless someone put setlocale()  either in run-tests or outside of libuv but I think it is better to make uv_dlerror() be independent of locale because people can report and share its error message regardless of their native language.
How about this one? f953faf8b70aa615927ede2b67d19650ee82cf17

File: src/win/dl.c [hunk 2]
```diff
@@ -61,3 +61,17 @@ uv_err_t uv_dlsym(uv_lib_t library, const char* name, void** ptr) {
   *ptr = (void*) proc;
   return uv_ok_;
 }
+
+
+const char *uv_dlerror(uv_lib_t library) {
+  char* buf = NULL;
+  FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM |
+                 FORMAT_MESSAGE_IGNORE_INSERTS, NULL, GetLastError(),
```
Comment (piscisaureus) [inline src/win/dl.c]: You should store GetLastError() in a thread-local after uv_dlopen(). Otherwise the behaviour will not be the same across platforms.

**Comment by shigeki** [comment]
@bnoordhuis  Very sorry for my typo as I made my check at the another branch and did not merge it. I apologize for bothering you to review my code in many times. 
@piscisaureus Thank you for your comment. I was not aware of it but found it on the top of the MSDN description of GetLastError().  Fix was made in e082861e03d0190755906f530c71556332b09817 , please review it.

**Comment by bnoordhuis** [comment]
@shigeki: Don't worry about it. I get paid to review patches, you know. :-)

**Comment by shigeki** [comment]
@bnoordhuis @piscisaureus Thanks. I owe you very much. I have to pay to Cloud9 :-) 
Fixed in ffafc3d7109d2bfcd1adf8dcd9a247ce7dcffe16

**Comment by bnoordhuis** [comment]
Thanks Shigeki, merged in b55801f.

**Comment by shigeki** [comment]
Thanks Ben. After you update libuv in node, I will submit a new PR of process.dlopen()

**Pull request closed without merging**